samples/rtl/rtl_tasklet_scalar.py

```diff
@@ -25,51 +25,75 @@
+        -->|     s_axis_{input}_tvalid   reg m_axis_{output}_tvalid |-->
+        -->|     s_axis_{input}_tdata    reg m_axis_{output}_tdata  |-->
+        <--| reg s_axis_{input}_tready       m_axis_{output}_tready |<--
+        -->|     s_axis_{input}_tkeep    reg m_axis_{output}_tkeep  |-->
+        -->|     s_axis_{input}_tlast    reg m_axis_{output}_tlast  |-->
+           |                                                        |
+           |--------------------------------------------------------|
     */
 
-    typedef enum [1:0] {READY, BUSY, DONE} state_e;
+    typedef enum logic [1:0] {READY, BUSY, DONE} state_e;
     state_e state;
 
-    always@(posedge clk_i) begin
-        if (rst_i) begin // case: reset
-            b <= 0;
-            ready_o <= 1'b1;
+    always@(posedge ap_aclk) begin
+        if (ap_areset) begin // case: reset
+            m_axis_b_tdata <= 0;
+            s_axis_a_tready <= 1'b1;
             state <= READY;
-        end else if (valid_i && state == READY) begin // case: load a 
-            b <= a;
-            ready_o <= 1'b0;
+        end else if (s_axis_a_tvalid && state == READY) begin // case: load a
+            m_axis_b_tdata <= s_axis_a_tdata;
+            s_axis_a_tready <= 1'b0;
             state <= BUSY;
-        end else if (b < 100) // case: increment counter b
-            b <= b + 1;
-        else
-            b <= b;
+        end else if (m_axis_b_tdata < 100) // case: increment counter b
+            m_axis_b_tdata <= m_axis_b_tdata + 1;
+        else begin
+            m_axis_b_tdata <= m_axis_b_tdata;
             state <= DONE;
-    end    
+        end
+    end
 
-    assign valid_o = (b >= 100) ? 1'b1:1'b0; 
+    assign m_axis_b_tvalid = (m_axis_b_tdata >= 100) ? 1'b1:1'b0;
     ''',
                             language=dace.Language.SystemVerilog)
 
 # add input/output array
 A = state.add_read('A')
 B = state.add_write('B')
 
+mentry, mexit = state.add_map('rtl_map', dict(i='0:1'))
+
 # connect input/output array with the tasklet
-state.add_edge(A, None, tasklet, 'a', dace.Memlet.simple('A', '0'))
-state.add_edge(tasklet, 'b', B, None, dace.Memlet.simple('B', '0'))
+state.add_memlet_path(A,
+                      mentry,
+                      tasklet,
+                      dst_conn='a',
+                      memlet=dace.Memlet.simple('A', '0'))
+state.add_memlet_path(tasklet,
+                      mexit,
+                      B,
+                      src_conn='b',
+                      memlet=dace.Memlet.simple('B', '0'))
 
 # validate sdfg
 sdfg.validate()
 
+#from dace.transformation.dataflow import StreamingMemory
+#from dace.transformation.interstate import FPGATransformState
+from dace.transformation.dataflow import TrivialMapElimination
+#sdfg.apply_transformations(FPGATransformState)
+#sdfg.apply_transformations_repeated(StreamingMemory,
+#                                    dict(storage=dace.StorageType.FPGA_Local))
```

Comment:
Cleanup

Reply:
Fixed in https://github.com/carljohnsen/dace/commit/11984d3c2c997860d2124ef82133077ccd84e753